app/controllers/AuditController.scala

```diff
@@ -88,8 +80,9 @@ class AuditController @Inject() (implicit val env: Environment[User, SessionAuth
           case None =>
             WebpageActivityTable.save(WebpageActivity(0, user.userId.toString, ipAddress, "Visit_Audit", timestamp))
 
-            val task: NewTask = if (region.isDefined) AuditTaskTable.selectANewTaskInARegion(region.get.regionId, user.userId)
-            else AuditTaskTable.selectANewTask(user.userId)
+            val task: NewTask =
+              if (region.isDefined) AuditTaskTable.selectANewTaskInARegion(region.get.regionId, user.userId)
```

Comment:
Here, in what conditions is the region not defined? 

Also, when the user finishes a neighborhood and clicks "Audit Another Neighborhood", at this point, it always goes to the else condition (where a street with highest priority is selected and the corresponding region is then assigned to the user). Even though its correct, but I am not sure why. Because region is always defined (since a user is currently assigned a region - the one they finish). An example is below:

![image](https://user-images.githubusercontent.com/2873216/36942115-2c7a4bee-1f20-11e8-802c-3cebaa47ea67.png)

The output of the print statements is below:

![image](https://user-images.githubusercontent.com/2873216/36942118-3b927b06-1f20-11e8-8574-c6bd1f80717a.png)

This is right after the user clicks Audit another neighborhood button when they finish a neighborhood on their own, and the page reloads. Its correct behavior however, I am not sure how its working. Can you explain how?

Reply:
>Here, in what conditions is the region not defined?

There aren't any cases where it shouldn't be defined. It is probably a redundant check. I would be okay with removing it.

Reply:
>Also, when the user finishes a neighborhood and clicks "Audit Another Neighborhood", at this point, it always goes to the else condition (where a street with highest priority is selected and the corresponding region is then assigned to the user). Even though its correct, but I am not sure why. Because region is always defined (since a user is currently assigned a region - the one they finish). An example is below:

Ahh, yes this should probably be changed, but here is how it works: If you follow the call to `AuditTaskTable.selectANewTaskInARegion(region.get.regionId, user.userId)`, if the user has completed all the tasks in that region, this function falls back to a function `selectANewTask(user: UUID)`, which picks the highest priority task across all of DC that the user has not audited. So reassigning the user to the new region is actually happening in the task selection function! Not ideal.

Reply:
Ah yes! I was assuming it was going via else condition. That's why it didn't make any sense. Thanks for the clarification. And yes, the region assignment should be done outside of that function.